### PR TITLE
waypoint up in the browser

### PR DESCRIPTION
**Pull request by gregone**
Fixes: #1238 

This PR basically reopens https://github.com/hashicorp/waypoint/pull/1500 with a few modifications.Because of some technical constraints, the "Waypoint Up" button will not be available to everyone. 
Users will only see it if: 
- they have a Git `DataSource` set up
- `DataSourcePoll` is not enabled (otherwise "Up" automatically runs)
- there are no existing releases

Example (fils to deploy for other reasons, but you get the idea): 

https://user-images.githubusercontent.com/1416421/134381988-41e22f32-e19e-438d-9698-4459b5955868.mov

### CODE REVIEW

File: ui/app/routes/workspace/projects/project/app.ts [hunk 1]
```diff
@@ -56,6 +56,7 @@ export default class App extends Route {
     appRef.setProject(proj.name);
 
     return hash({
+      project: proj,
```
Comment (jgwhite) [inline ui/app/routes/workspace/projects/project/app.ts:59]: Can we add this field to the `Model` interface defined at the top of the file?

**Review by jgwhite — CHANGES_REQUESTED** [review]
This is looking great 👍 I’m just testing it locally to be double-sure.

Aside from the one inline comment, could I also suggest we rebase and squash this?

File: ui/app/routes/workspace/projects/project/app.ts (hunk 1 above)
Comment (gregone) [reply ui/app/routes/workspace/projects/project/app.ts:60]: done

Comment (gregone) [reply ui/app/routes/workspace/projects/project/app.ts:60]: I'll use the `squash and merge` button on GH

**Review by jgwhite — APPROVED** [review]

**Pull request merged**